code/Report.php

```diff
@@ -97,6 +99,13 @@ class Report extends ViewableData
         SideReportWrapper::class,
     ];
 
+    public function __construct()
+    {
+        $this->params = isset($_REQUEST['filters']) ? $_REQUEST['filters'] : array();
```

Comment:
This change doesn't actually implement the given TODO

```php
// TODO Remove coupling with global state
```

This in fact makes this code riskier; It relies on the global state a lot earlier (object construction, not at render time). This makes testing more susceptible to global state change.

Reply:
@tractorcow The goal of this commit wasn't to implement the TODO and I can put it back.

Reply:
This still leaves this concern;

> This in fact makes this code riskier; It relies on the global state a lot earlier (object construction, not at render time). This makes testing more susceptible to global state change.